snap/README.md

```diff
@@ -0,0 +1,164 @@
+and then tries again up to a maximum duration (default: 60s). These values can be overridden with the following commands:
+    
+To change the maximum duration, use the following command:
+
+```bash
+$ sudo snap set edgex-app-rfid-llrp-inventory startup-duration=60
+```
+
+To change the interval between retries, use the following command:
+
+```bash
+$ sudo snap set edgex-app-rfid-llrp-inventory startup-interval=1
+```
+
+The service can then be started as follows. The "--enable" option
+ensures that as well as starting the service now, it will be automatically started on boot:
+
+```bash
+$ sudo snap start --enable edgex-device-rfid-llrp.device-rfid-llrp
+```
+
+### Aliases setup
+
+The `AppConfig.Aliases` setting needs to be provided for the service to work.  See [Setting the Aliases](https://github.com/edgexfoundry/app-rfid-llrp-inventory#setting-the-aliases).
+
+This can either be by
+
+1. using a content interface to provide a `configuration.toml` file with the correct aliases, or
+
+2. during development, set the values manually in Consul. 
+
+
+### Using a content interface to set device configuration
+
+The `device-config` content interface allows another snap to seed this snap with configuration directories under `$SNAP_DATA/config/app-rfid-llrp-inventory`.
+
+Note that the `device-config` content interface does NOT support seeding of the Secret Store Token because that file is expected at a different path.
+
+Please refer to [edgex-config-provider](https://github.com/canonical/edgex-config-provider), for an example and further instructions.
+
+
+### Rich Configuration
+While it's possible on Ubuntu Core to provide additional profiles via gadget 
+snap content interface, quite often only minor changes to existing profiles are required. 
+
+These changes can be accomplished via support for EdgeX environment variable 
+configuration overrides via the snap's configure hook.
+If the service has already been started, setting one of these overrides currently requires the
+service to be restarted via the command-line or snapd's REST API. 
+If the overrides are provided via the snap configuration defaults capability of a gadget snap, 
+the overrides will be picked up when the services are first started.
+
+The following syntax is used to specify service-specific configuration overrides:
+
+
+```
+env.<stanza>.<config option>
+```
+For instance, to setup an override of the service's Port use:
+```
+$ sudo snap set edgex-device-rfid-llrp env.service.port=2112
+```
+And restart the service:
+```
+$ sudo snap restart edgex-device-rfid-llrp.device-rfid-llrp
+```
+
+## Service Environment Configuration Overrides
+
+### [Service]
+|snap set|configuration.yaml setting|
+|---|---|
+|env.service.boot-timeout|Service.BootTimeout|
+|env.service.health-check-interval|Service.HealthCheckInterval|
+|env.service.host|Service.Host|
+|env.service.server-bind-addr|Service.ServerBindAddr|
+|env.service.port|Service.Port|
```

Comment:
This doesn't work:

```
$ sudo snap set edgex-app-rfid-llrp-inventory env.service.port=55555
$ sudo snap restart edgex-app-rfid-llrp-inventory
```

Logs:
```
$ sudo snap logs -f -n 100 edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory | grep localhost
...
2022-02-17T18:18:07+01:00 edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory[62516]: level=INFO ts=2022-02-17T17:18:07.503249243Z app=app-rfid-llrp-inventory source=server.go:156 msg="Starting HTTP Web Server on address localhost:59711"
...
2022-02-17T18:28:57+01:00 edgex-app-rfid-llrp-inventory.app-rfid-llrp-inventory[65781]: level=INFO ts=2022-02-17T17:28:57.713082759Z app=app-rfid-llrp-inventory source=server.go:156 msg="Starting HTTP Web Server on address localhost:59711"
```

Tried `server-bind-addr` too; it had no effect.